engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/interop/syntax/MethodDispatchNode.java

```diff
@@ -53,7 +51,7 @@ public static MethodDispatchNode build(Language language) {
    * @param frame current execution frame.
    * @return the result of converting input into a string.
    */
-  @Specialization(guards = "symbol == cachedSymbol")
+  @Specialization(guards = "symbol.getScope() == cachedSymbol.getScope()")
   public Stateful run(
       VirtualFrame frame,
       @Bind("getSymbol(frame)") UnresolvedSymbol symbol,
```

Comment:
As we discussed, this change may be slightly misleading, since now we have no guarantee that `symbol == cachedSymbol` and we pass `cachedSymbol` to `buildToArray`.
As we discussed, it is correct, because `buildToArray` only uses `argument.getScope()`, and these are guaranteed to be equal.

But maybe it could become more readable if `buildToArray` was changed to just take the `Scope` it needs instead of the whole `UnresolvedSymbol`. Then the call here would be `buildToArray(cachedSymbol.getScope())` (or even maybe the scope could be what is cached? I'm not sure if that is allowed), but regardless, its correctness would be much more obvious since we would see at callsite that just `symbol.getScope() == cachedSymbol.getScope()` is needed.